src/meshes/p4est_mesh_view.jl

```diff
@@ -0,0 +1,209 @@
+@muladd begin
+#! format: noindent
+
+mutable struct P4estMeshView{NDIMS, RealT <: Real, IsParallel, P, Ghost, NDIMSP2,
+                             NNODES} <:
+               AbstractMesh{NDIMS}
+    # Attributes from the original P4est mesh
+    p4est       :: P # Either PointerWrapper{p4est_t} or PointerWrapper{p8est_t}
+    is_parallel :: IsParallel
+    ghost       :: Ghost # Either PointerWrapper{p4est_ghost_t} or PointerWrapper{p8est_ghost_t}
+    # Coordinates at the nodes specified by the tensor product of `nodes` (NDIMS times).
+    # This specifies the geometry interpolation for each tree.
+    tree_node_coordinates::Array{RealT, NDIMSP2} # [dimension, i, j, k, tree]
+    nodes::SVector{NNODES, RealT}
+    boundary_names::Array{Symbol, 2}      # [face direction, tree]
+    current_filename::String
+    unsaved_changes::Bool
+    p4est_partition_allow_for_coarsening::Bool
+
+    # Attributes pertaining the views.
+    parent::P4estMesh{NDIMS, RealT}
+    indices_min::NTuple{NDIMS, Int}
```

Comment:
I assume `indices_min` and `indices_max` indicate the mesh sub-region associated to the `P4estMeshView` instance?

This only works for meshes structured in an `nx` x `ny` fashion. It is not really generic to other unstructured meshes. A more flexible approach would be to store the start and end index of the global coarse mesh array of the p4est instance. Or, alternatively, have an array with integer flags indicating which coarse cell resp. tree in the forest is associated with which `P4estViewMesh` resp. semi-discretization.

Reply:
You are right. This is a first setup towards coupling. In this PR we want to limit ourselves to a simple domain decomposition. Later we will make it more flexible.

Reply:
Fair enough!